### PR TITLE
Bug 2294 - Tests for routes

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

File: cornfield/test/mock.user.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+function generateMockData(id) {
+  id = id || (Math.random()*1000000000).toFixed(0);
+
+  return {
+    _id: id,
+    data: JSON.stringify({
+      hello: "world",
+      adventure: "bill & ted's"
```
Comment (mjschranz) [inline cornfield/test/mock.user.js:8]: Something tells me you want to watch this movie.

File: cornfield/test/routes.index.test.js [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+      t.equal(res.body.error, "okay", "json returns okay");
+
+      t.end();
+    });
+});
+
+/*******************
+ * DANGER ZONE     *
+ * HERE BE DRAGONS *
+ *******************/
```
Comment (mjschranz) [inline cornfield/test/routes.index.test.js:121]: Uhhh, what?

File: cornfield/test/routes.index.test.js [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+test("create project with error", function(t) {
```
Comment (mjschranz) [inline cornfield/test/routes.index.test.js:123]: should this be renamed to with no id instead?

File: cornfield/test/routes.index.test.js [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+      t.end();
+    });
+});
+
+test("create project with no data", function(t) {
+  mockUser.error = false;
+
+  var mockData = mockUser.generateMockData(1234);
+  delete mockData._id;
```
Comment (mjschranz) [inline cornfield/test/routes.index.test.js:145]: Technically don't need to delete the ID do we?

File: cornfield/test/routes.index.test.js [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+
+  request(app)
+    .post("/api/project/1234")
+    .send(mockData)
+    .end(function(err, res) {
+      t.equal(res.statusCode, 200, "status code is 200");
+      t.equal(res.type, "application/json", "response type is json");
+      t.equal(res.body.error, "okay", "status is okay");
+      mockData.data = JSON.stringify(mockData.data);
+      t.deepEqual(res.body.project, mockData);
```
Comment (mjschranz) [inline cornfield/test/routes.index.test.js]: No message?

File: cornfield/test/routes.index.test.js [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+    .end(function(err, res) {
+      t.equal(res.statusCode, 200, "status code is 200");
+      t.equal(res.type, "application/json", "response type is json");
+
+      // This is very obtuse...
+      var mockData = mockUser.generateMockData(1234);
+      mockData.data = JSON.parse(mockData.data);
+      mockData.data.name = "Remix of " + mockData.name;
+      mockData = mockData.data;
+      t.deepEqual(res.body, mockData);
```
Comment (mjschranz) [inline cornfield/test/routes.index.test.js]: Again, no message?

File: cornfield/test/routes.index.test.js [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+  delete mockData._id;
+  delete mockData.data;
+
+  request(app)
+    .post("/api/project/1234")
+    .send(mockData)
+    .end(function(err, res) {
+      t.equal(res.statusCode, 500, "status code is 500");
+      t.equal(res.type, "application/json", "response type is json");
+      t.equal(res.body.error, "not enough parameters to update");
```
Comment (mjschranz) [inline cornfield/test/routes.index.test.js]: Message please!

File: cornfield/test/routes.index.test.js [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+  var mockData = mockUser.generateMockData(1234);
+  delete mockData._id;
+
+  request(app)
+    .post("/api/project/1234")
+    .send(mockData)
+    .end(function(err, res) {
+      t.equal(res.statusCode, 500, "status code is 500");
+      t.equal(res.type, "application/json", "response type is json");
+      t.equal(res.body.error, "mock error");
```
Comment (mjschranz) [inline cornfield/test/routes.index.test.js]: Message Please

File: cornfield/test/routes.index.test.js [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+      t.equal(res.type, "application/json", "response type is json");
+
+      // This is very obtuse...
+      var mockData = mockUser.generateMockData(1234);
+      mockData.data = JSON.parse(mockData.data);
+      mockData.data.name = mockData.name;
+      mockData.data.projectID = mockData._id;
+      mockData.data.author = mockData.author;
+      mockData = mockData.data;
+      t.deepEqual(res.body, mockData);
```
Comment (mjschranz) [inline cornfield/test/routes.index.test.js]: Message Please!

File: cornfield/test/routes.index.test.js (hunk 3 above)
Comment (jbuck) [inline cornfield/test/routes.index.test.js:123]: "new" projects shouldn't have an id

File: cornfield/test/routes.index.test.js (hunk 4 above)
Comment (jbuck) [inline cornfield/test/routes.index.test.js:145]: for the create tests, yes

**Pull request merged**